src/pages/steps/AdditionalInformationStep/FormScore.tsx

```diff
@@ -128,21 +129,51 @@ const scoreWeightMapping: Weights = {
   },
 };
 
+const organizerScoreWeightMapping: Weights = {
+  name: {
+    weight: 20,
+    mandatory: true,
+  },
+  url: {
+    weight: 20,
+    mandatory: true,
+  },
+  contact_info: {
+    weight: 20,
+    mandatory: false,
+  },
+  description: {
+    weight: 15,
+    mandatory: false,
+  },
+  media: {
+    weight: 15,
+    mandatory: false,
+  },
+  location: {
+    weight: 10,
+    mandatory: false,
+  },
+};
+
 type Props = {
   offerId: string;
   scope: Scope;
   completedFields: Record<Field, boolean>;
 };
 
-const getScopeWeights = (scope: Scope): Weights => {
-  let weights = scoreWeightMapping;
-  if (scope === ScopeTypes.ORGANIZERS) {
-    weights.location.weight = 10;
-    weights.description.weight = 15;
-  }
+type EntityWithMedia =
+  | (Offer & { images: undefined })
+  | (Organizer & {
+      terms: undefined;
+      mediaObject: undefined;
+      videos: undefined;
+    });
```

Comment:
Is your intention to show offer can have images or no images, etc?

Because you can make the properties optional on the `Offer` and `Organizer` if they're not correct.

Reply:
No all the properties are correct, but their presence/name/required is not the same for offers as for organizers. Offers have `mediaObject` while Organizer uses `images`, in the same way organizers don't have `terms`. This type is just a "trick" to make not have to `expect-error` 20 lines in the component cause we do a lot of checks to properties that only exist in one or the other :/

Reply:
I think the correct handling of this could perhaps be having a `isOrganizer` [type predicate function](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) which would be able to make Typescript differentiate which object we're dealing with, as currently even with coalesce operators TS complains since it's more about the existence of the property on the objects than them being defined